### ISSUE TITLE
Embedded xml could gen non suppressible/fixable warnings

### ISSUE THREAD

**Issue by tlakollo** [mono/linker#1179]
Remove ResolveStep (#1169) changes AddUnresolveMarker calls for a warning message, this could lead to warnings that cannot be deleted in the case an XML file comes embedded in the assembly. 
Example:
You have an application that includes a nugget package, no source code for it. and contains an embedded XML on it with a bug/mismatch and talks about an unexisting member. If it warns there would be no way to get rid of the warning.
XmlAnnotation (#1176) mimics this same behavior

**Comment by vitek-karas** [comment]
/cc @mateoatr 

@marek-safar Would you know if it's important to have a solution for this?
I'm sort of split on this:
* On the one hand if we report these things as just informational messages it's likely that almost nobody will notice them. So having them show up as warnings is in my mind a good thing.
* On the other hand if my app uses a NuGet which happens to have a wrong embedded XML in it, there's no way for me to suppress the warning - which is bad.

Do we have experience with this in Xamarin?

**Comment by marek-safar** [comment]
I think we should have a solution for it as the embedded XML are in most cases the only solution how to reliably ship the data. We don't issue any warnings in Xamarin today (errors only) but I think this can happen on existing code.

The easy solution would be to use or extend [Scope](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.suppressmessageattribute.scope?view=netcore-3.1#remarks) property which we'll have to support most likely anyway.

**Comment by vitek-karas** [comment]
Just to make sure I understand: By "solution" you mean to have a way to suppress warnings coming from parsing embedded XML files, right?

As you mention we should support `Scope`, unfortunately we would probably have to extend it to some degree - I don't know if it can be used to suppress warnings in a different module/assembly. But if yes - that would be the right way to do so.

**Comment by marek-safar** [comment]
Yes, `Scope = "resource"` should suppress warning related to embedded data because we would use the "resource" as the source location, right?

We could extend the name to make it more obvious or even support embedded resource name as the scope.

**Comment by vitek-karas** [comment]
I missed the `resource` value, you're right. Now if only this works cross assembly (attribute in one assembly targeting a different assembly). If not we would have to rely on the XML attribute format - would work, but still.

**Comment by marek-safar** [comment]
I think we issue most XML warnings at load time so that should work fine.
